examples/preprocessing/plot_scaling_importance.py

```diff
@@ -1,106 +1,149 @@
 
-print("\nPrediction accuracy for the standardized test dataset with PCA")
-print(f"{accuracy_score(y_test, pred_test_std):.2%}\n")
 
-# Extract PCA from pipeline
-pca = unscaled_clf.named_steps["pca"]
-pca_std = std_clf.named_steps["pca"]
+def fit_and_plot_model(X_plot, y, clf, ax):
+    clf.fit(X_plot, y)
+    disp = DecisionBoundaryDisplay.from_estimator(
+        clf,
+        X_plot,
+        response_method="predict",
+        alpha=0.5,
+        ax=ax,
+    )
+    disp.ax_.scatter(X_plot[:, 0], X_plot[:, 1], c=y, s=20, edgecolor="k")
+    disp.ax_.set_xlim((X_plot[:, 0].min(), X_plot[:, 0].max()))
+    disp.ax_.set_ylim((X_plot[:, 1].min(), X_plot[:, 1].max()))
+    return disp.ax_
 
-# Show first principal components
-print(f"\nPC 1 without scaling:\n{pca.components_[0]}")
-print(f"\nPC 1 with scaling:\n{pca_std.components_[0]}")
 
-# Use PCA without and with scale on X_train data for visualization.
+fig, (ax1, ax2) = plt.subplots(ncols=2, figsize=(12, 6))
+
+fit_and_plot_model(X_plot, y, clf, ax1)
+ax1.set_xlabel("proline")
+ax1.set_ylabel("hue")
+ax1.set_title("KNN w/o scaling")
+
+fit_and_plot_model(X_plot_scaled, y, clf, ax2)
+ax2.set_xlabel("scaled proline")
+ax2.set_ylabel("scaled hue")
+ax2.set_title("KNN with scaling")
+
+plt.show()
+
+# %%
+# Here the desicion boundary shows that fitting scaled or non-scaled data lead
+# to completely different KNeighbors models. The reason is that the variable
+# "proline" has values which vary between 0 and 1,000; whereas the variable
+# "hue" varies between 1 and 10. Because of this, distances between samples are
+# mostly impacted by differences in values of "proline", while values of the
+# "hue" will be comparatively ignored. If one uses
+# :class:`~sklearn.preprocessing.StandardScaler` to normalize this database,
+# both scaled values lay approximately between -3 and 3 and the neighbors
+# structure will be impacted more or less equivalently by both variables.
+#
+# Effect of rescaling on a PCA dimensional reduction
+# ==================================================
+#
+# Dimensional reduction using PCA consists of finding the features that maximize
+# the variance. If one feature varies more than another only because of their
+# respective scales, :class:`~sklearn.decomposition.PCA` would determine that
+# such feature dominates the direction of the principal components.
+#
+# We can inspect the first principal components on the full dataset:
+
+from sklearn.decomposition import PCA
+
+pca = PCA(n_components=2).fit(X_train)
+scaled_pca = PCA(n_components=2).fit(scaled_X_train)
 X_train_transformed = pca.transform(X_train)
+X_train_std_transformed = scaled_pca.transform(scaled_X_train)
 
-scaler = std_clf.named_steps["standardscaler"]
-scaled_X_train = scaler.transform(X_train)
-X_train_std_transformed = pca_std.transform(scaled_X_train)
+print(f"\nPC 1 without scaling:\n{pca.components_[0]}")
+print(f"\nPC 1 with scaling:\n{scaled_pca.components_[0]}")
```

Comment:
Since we want to look at the `feature #13`, I think it would be easier to show to the output of a `DataFrame` with the column/index name. It will ease the reading.

Reply:
We could even used the original feature names then.